### PR TITLE
Make prefetching work with `usePaginatedQuery` and `useInfiniteQuery`

### PR DESCRIPTION
**Pull request by beerose**
Closes: https://github.com/blitz-js/blitz/discussions/2839

Fixed:

* Using `usePaginatedQuery` with dehydrated state
* Using `useInfiniteQuery` with dehydrated state

## Bug Checklist

- [ ] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)

## Feature Checklist

- [ ] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)
- [ ] Documentation added/updated (submit PR to [blitzjs.com repo `canary` branch](https://github.com/blitz-js/blitzjs.com/tree/canary))

### CODE REVIEW

File: test/integration/queries/pages/dehydrated-state.tsx [hunk 1]
```diff
@@ -27,11 +26,7 @@ function Content() {
 }
 
 function DehydratedState() {
-  return (
-    <Suspense fallback="Loading ...">
-      <Content />
-    </Suspense>
-  )
+  return <Content />
```
Comment (flybayer) [inline test/integration/queries/pages/dehydrated-state.tsx:29]: I'm pretty sure suspense boundary is needed here for client navigation

Comment (flybayer) [reply test/integration/queries/pages/dehydrated-state.tsx:29]: actually maybe not..

Comment (beerose) [reply test/integration/queries/pages/dehydrated-state.tsx:29]: I just checked to be sure, and client navigation works fine without suspense

File: nextjs/packages/next/data-client/react-query-utils.ts [hunk 2]
```diff
@@ -125,6 +125,23 @@ export function getQueryKey<TInput, TResult, T extends AsyncFunc>(
   return getQueryKeyFromUrlAndParams(sanitizeQuery(resolver)._routePath, params)
 }
 
+export function getInfiniteQueryKey<TInput, TResult, T extends AsyncFunc>(
+  resolver: T | Resolver<TInput, TResult> | RpcClient<TInput, TResult>,
+  params?: TInput
+) {
+  if (typeof resolver === 'undefined') {
+    throw new Error(
+      'getQueryKey is missing the first argument - it must be a resolver function'
```
Comment (beerose) [inline nextjs/packages/next/data-client/react-query-utils.ts:134]:
```suggestion
      'getInfiniteQueryKey is missing the first argument - it must be a resolver function'
```

**Review by flybayer — APPROVED** [review]

**Pull request merged**